internal/clients/sources/sources_client.go

```diff
@@ -0,0 +1,153 @@
+package sources
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/RHEnVision/provisioning-backend/internal/cache"
+	"github.com/RHEnVision/provisioning-backend/internal/config"
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+	"github.com/RHEnVision/provisioning-backend/internal/headers"
+	"github.com/RHEnVision/provisioning-backend/internal/parsing"
+)
+
+type SourcesClient struct {
+	client *ClientWithResponses
+}
+
+func init() {
+	GetSourcesClientV2 = newSourcesClient
+}
+
+func newSourcesClient(ctx context.Context) (ClientV2, error) {
+	c, err := NewClientWithResponses(config.Sources.URL)
+	if err != nil {
+		return nil, err
+	}
+	return &SourcesClient{client: c}, nil
+}
+
+func (c *SourcesClient) ListProvisioningSources(ctx context.Context) (*[]Source, error) {
+	ctxval.Logger(ctx).Info().Msg("Listing provisioning sources")
+	appTypeId, err := c.GetProvisioningTypeId(ctx)
+	if err != nil {
+		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to get provisioning type id")
+		return nil, fmt.Errorf("failed to get provisioning app type: %w", err)
+	}
+	resp, err := c.client.ListApplicationTypeSourcesWithResponse(ctx, appTypeId, &ListApplicationTypeSourcesParams{}, headers.AddIdentityHeader)
+	if err != nil {
+		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to fetch ApplicationTypes from sources")
+		return nil, fmt.Errorf("failed to get ApplicationTypes: %w", err)
+	}
+	statusCode := resp.StatusCode()
+
+	if parsing.IsHTTPNotFound(statusCode) {
+		return nil, NewNotFoundError(ctx, "sources not found")
+	}
+	if !parsing.IsHTTPStatus2xx(statusCode) {
+		ctxval.Logger(ctx).Warn().Msgf("Sources replied with unexpected status while fetching sources of application type: %v", statusCode)
```

Comment:
%d would probably be a better fit given we know it's an integer, but this is ok as well :)
```suggestion
		ctxval.Logger(ctx).Warn().Msgf("Sources replied with unexpected status while fetching sources of application type: %d", statusCode)
```

Reply:
the linter didn't like %d for some reason so I changed it to %v

Reply:
ok, I thought we had %s there before... but if %d doesn't work let's go with %v :+1: